Source/GSGuiPrivate.h

```diff
@@ -103,5 +104,24 @@ static inline CGFloat GSRoundTowardsNegativeInfinity(CGFloat x)
   return ceil(x - 0.5);
 }
 
+#if !defined(GS_DECODER)
+#define GS_DECODER(type) \
+static inline void decode_##type(NSCoder *coder, type *value) \
+{ \
+[coder decodeValueOfObjCType: @encode(type) at: value]; \
+}
+
```

Comment:
While I don't believe it's as much of a risk for keyed archives, should there be an equivalent set of macros for them?